src/main/java/dinkplugin/notifiers/DeathNotifier.java

```diff
@@ -6,19 +6,28 @@
 import dinkplugin.message.NotificationBody;
 import dinkplugin.message.NotificationType;
 import dinkplugin.notifiers.data.DeathNotificationData;
-import net.runelite.api.*;
+import net.runelite.api.Actor;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.Player;
+import net.runelite.api.Prayer;
 import net.runelite.api.events.ActorDeath;
+import net.runelite.api.events.InteractingChanged;
 import org.apache.commons.lang3.tuple.Pair;
 
 import javax.inject.Inject;
+import java.lang.ref.WeakReference;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
-import java.util.Objects;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class DeathNotifier extends BaseNotifier {
 
+    private WeakReference<Actor> lastTarget = new WeakReference<>(null);
```

Comment:
This member might seem out of place of a DeathNotifier, so maybe a little comment why we're storing this makes sense